src/types/UiaTextRangeBase.cpp

```diff
@@ -1318,6 +1336,15 @@ const Viewport UiaTextRangeBase::_getBufferSize() const noexcept
     return Viewport::FromDimensions({ 0, 0 }, width, height);
 }
 
+const til::point UiaTextRangeBase::_getDocumentEnd() const noexcept

```

Comment:
Idea: I could cache `_getDocumentEnd()`, and in the scope exit for (un)locking the console, I could reset the cache

Worth pursuing?

Reply:
I might add the description from the body of this PR here, since this is kinda a load-bearing function

>  We consider the "document end" to be the line beneath the cursor or last legible character (whichever is further down). In the event where the last legible character is on the last line of the buffer, we use the "end exclusive" position (left-most point on a line one past the end of the buffer).